umbral/signing.py

```diff
@@ -0,0 +1,101 @@
+from cryptography.exceptions import InvalidSignature
+
+from umbral.config import default_curve
+from umbral.keys import UmbralPublicKey
+from cryptography.hazmat.primitives.asymmetric.utils import decode_dss_signature, encode_dss_signature
+
+from umbral.keys import UmbralPrivateKey
+from cryptography.hazmat.primitives.asymmetric import ec
+from cryptography.hazmat.primitives import hashes
+
+from umbral.utils import get_curve_keysize_bytes
+
+_BLAKE2B = hashes.BLAKE2b(64)
+
+
+class Signature(object):
+    """
+    The Signature object allows signatures to be made and verified.
+    """
+
+    def __init__(self, r: int, s: int):
```

Comment:
If we verify using standard functions which take der-encoded bytes, why do we keep integers for `r` and `s` at all?

Reply:
So that we also have the option of serializing to a fixed-length format.

Reply:
Ah, makes sense

Reply:
I don't know if I'd go that far.  ;-)

Reply:
I know, but it also allows to user `(r,s)`. This is from my fork:
```
from cryptography.hazmat.primitives.asymmetric.utils import decode_dss_signature, encode_dss_signature

signature = self._private_key.sign(data, ec.ECDSA(hashes.SHA256()))
(r, s) = decode_dss_signature(signature)
```

Reply:
Right - we went around and around about this when we first adopted `Signature`, remember?

`Signature` is really just a machine for keeping internal state as `(r, s)` and encoding to / from DER for use in cryptography.io.